src/mockServiceWorker.js

```diff
@@ -21,8 +21,13 @@ self.addEventListener('activate', async function (event) {
 })
 
 self.addEventListener('message', async function (event) {
+  if (!event.source.id) return
   const clientId = event.source.id
+
+  if (!event.currentTarget.clients) return
   const client = await event.currentTarget.clients.get(clientId)
+
+  if (!client) return
```

Comment:
I believe this is the actual fix for the issue: if we didn't find the client by the given ID that's controlled by MSW's worker, then do not attempt to handle the incoming message.